### PR TITLE
Added loading indicator to modules page

### PR DESCRIPTION
**Pull request by xbili**
For [#95]. A temporary fix for [#156] as well.

I don't think this is the best way to add in a spinner, apparently people have been pushing for Marionette's CollectionView to come with a LoadingView [here] as well.

Any better way to do this? 

[#95]: https://github.com/nusmodifications/nusmods/issues/95
[#156]: https://github.com/nusmodifications/nusmods/issues/156
[here]: https://github.com/marionettejs/backbone.marionette/issues/510

### CODE REVIEW

File: app/scripts/common/templates/loading.hbs [hunk 1]
```diff
@@ -0,0 +1 @@
+<div class="loading"></div>
```
Comment (yangshun) [inline app/scripts/common/templates/loading.hbs]: Please add newline at the end of file

Comment (yangshun) [inline app/scripts/common/templates/loading.hbs]: It'd be good if we can namespace all our own classes by prefixing with `nm-`. Sorry I should come up with a style guide soon >_<

Comment (yangshun) [inline app/scripts/common/templates/loading.hbs]: Could you make it:
```
<div class="js-nm-loading nm-loading"></div>
```
instead? and use the `js-nm-loading` class in `LoadingView.js` and `nm-loading` in the CSS file.

**Comment by xbili** [comment]
I see, I'll namespace it first then!

File: app/scripts/common/templates/loading.hbs (hunk 1 above)
Comment (xbili) [inline app/scripts/common/templates/loading.hbs]: I see, I'll namespace it first then!

Comment (yangshun) [inline app/scripts/common/templates/loading.hbs]: This is a good read recommended for front end devs: http://philipwalton.com/articles/css-architecture/

Comment (xbili) [inline app/scripts/common/templates/loading.hbs]: Great article! Perfect for future reference as well.

So in this case we're using js-nm-loading for js manipulations and the other for styling only? That's really nice and neat :D

Made the changes!

Comment (yangshun) [inline app/scripts/common/templates/loading.hbs]: That's right, and `js-` prefixed selectors should not appear in the CSS file at all (although there are some existing there, which I should clean up). Awesome, lemme test it and I'll merge (:

**Comment by yangshun** [comment]
@xbili it works well! I'd want to modify its appearance though. I'll merge it and take over from here. Thanks! (:

**Comment by xbili** [comment]
Sure no problem! :)

**Pull request merged**